### ISSUE TITLE
Order of download list rearranges a few seconds after initial list display

### ISSUE THREAD

**Issue by kungfoolfighting** [piejanssens/premiumizer#77]
Hello!
I have the following issue:
Whenever I go to the home screen I am presented with a list of items that corresponds to the order of the list in the downloader view on premiumize.me. After a few seconds however, two older downloads are moved on top. This has already led to several accidental deletions of wrong files when this reordering happens just before I click to delete something.
If there is any information that I can supply to help you debug this, let me know!

**Comment by neox387** [comment]
Damn weird... there where more bugs however that are still in PR but don't think it has anything todo with this

What is the status of the items ? they downloading in cloud/local etc ?

they just move when going to the home screen or only after u delete something ?

It happens in another browser also ?

**Comment by kungfoolfighting** [comment]
I happens whenever I go to the home screen without having to click anything else.
I just checked and it happens both with firefox and with chromium.
I am running both premiumizer and the two browsers on Linux.
The status of the items is "Finished" and not downloaded or anything. I had added them manually a long time ago.
Perhaps it is worth noting that the current top download (before the reordering) is the Arthur C Clarke one, but its status is "Download queued" and it says "Loading name" and has no speed or progress.
I will verify if it also happens when this download does not exist.

**Comment by neox387** [comment]
Might be best to test again after next pr it might fix it there was a problem getting some cloud info.
But then there would be some exceptions in the log.  

So ye the loading name one maybe causes the issue :)

**Comment by kungfoolfighting** [comment]
Okay, I have played around with it a bit, and the behavior is really weird.
I did the following:
- I removed the download that was not working -> The reordering issue went away.
- I checked with firefox -> it would sometimes not load the list at all and the gear icon would constantly turn. Sometimes it would also list the list normally without the delayed reload.
- I checked the entire list and noticed that both don't display the last two downloads, which are the two elements that pop in after a few seconds with the stuck download present.
- I tried accessing the premiumizer via firefox without my reverse proxy in between and the list in firefox was complete; Also: The very second I accessed premiumizer without the reverse proxy in firefox, in premiumizer that was open in chromium immediately popped in the two missing downloads on top of the download list.
- Chromium loads the entire list as well when I dont go through the reverse proxy.

So to conclude: I think the reverse proxy might be set up wrong, but even then I have no idea what could cause this strange behavior, so perhaps it is a mixture of bad reverse proxy settings and some strange bug.
Here are the reverse proxy settings that I use:

```
    server {
        listen 80;
        server_name premiumizer.*;
        location / {
            proxy_pass http://homeserver:12000;
            proxy_set_header Host $http_host;
            proxy_set_header X-Real-IP $remote_addr;
            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
            proxy_set_header X-Forwarded-Proto $scheme;
        }
    }
```
I assume I am missing something here, since the settings you use for the reverse proxy with "/premiumizer/" have another section.

Thanks for all your help!

**Comment by neox387** [comment]
Yes the /socketio section should be needed also strange you get any list at all without it. Not sure if it will work when u keep it / instead if /prem for example. If u would use more stuff like sonarr etc with the reverse proxy you would need to split it up anyway.
